frontend/lib/util/strings.ts

```diff
@@ -5,3 +5,9 @@ export function ellipsis(text: string, maxLength: number = 20): string {
 
   return text.substring(0, maxLength - 3) + "...";
 }
+
+// Checks if the _whole_ string is numeric, returning it as a number
+// returns undefined when non-numeric characters are encountered
+export function parseIntStrict(text: string, radix: number = 10): number | undefined {
+  return /^\d+$/.test(text) ? parseInt(text, radix) : undefined;
```

Comment:
Do we have any idea of the performance implications of using a regex here? Regexes are not necessarily fast.

Reply:
No, not exactly. I guess it would partly depend on the browser's JavaScript engine. I went for the easiest implementation first, but I'm happy to write a more straightforward implementation here.

Reply:
Also, we need tests :)

Reply:
Also, with this regex the radix doesn't make sense for radices > 10 (e.g. "7a" is valid hex but doesn't match the regex).

If we set radix = 10, this might be an alternative without regexes:


```typescript
function parseInteger(str: string): number | undefined {
  const num = parseInt(str, 10);
  return !isNaN(num) && num.toString() === str ? num : undefined;
}

// Example usage:
console.log(parseInteger("123")); // 123
console.log(parseInteger("123.45")); // undefined
console.log(parseInteger("abc")); // undefined
console.log(parseInteger("123abc")); // undefined
```